prototyping/api_tables.ipynb

```diff
@@ -0,0 +1,613 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def request_text():\n",
+    "    'Returns raw card text data frame'\n",
+    "    df = pd.DataFrame(request('text')).set_index('id')\n",
+    "\n",
+    "    # Collapse columns raw and text, prioritizing raw\n",
+    "    df.loc[~df.raw.isna(), 'text'] = df.raw\n",
+    "    df.drop('raw', axis='columns', inplace=True)\n",
+    "\n",
+    "    return df\n",
+    "\n",
+    "raw_text = request_text()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get functions\n",
+    "\n",
+    "After requesting all JSONs, we have to filter and normalize the data, which in this case results in multiple dataframes.\n",
+    "\n",
+    "### Card information\n",
+    "\n",
+    "The main function in the following block is `get_card_information()`. It gets data from `raw_card` and combines with relevant text from `text`. Normalization results in the following data frames with respective schema:\n",
+    "\n",
+    "- `card`:\n",
+    "    - `card_id` (Integer PK): corresponding to `titleId` in JSON\n",
+    "    - `art_id` (Integer): art id in case I find a way to get images\n",
+    "    - `set_id` (String): three-letter set identifier\n",
+    "    - `title` (String): name of the card, necessarily unique\n",
+    "    - `rarity` (String): rarity of a card\n",
+    "    - `power` (String): creature power. Can be an integer, null or '*'\n",
+    "    - `toughness` (String): creature toughness. Can be an integer, null or '*'\n",
+    "    - `flavor` (String): flavor text, if a card has any\n",
+    "    - `is_legendary` (Boolean): flags a card is legendary\n",
+    "    - `is_token` (Boolean): flags a card token\n",
+    "    - `is_secondary_card` (Boolean): Flags a card secondary in an archetype\n",
+    "    - `is_rebalanced` (Boolean): Flags a digitally rebalanced card\n",
+    "- `card_type`:\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `type` (String PK): type according to magic rules\n",
+    "- `card_subtype`:\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `subtype` (String PK): subtype according to magic rules\n",
+    "- `card_cost`:\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `color` (String PK): six color identifiers according to magic rules\n",
+    "    - `cost` (Integer): amount of mana to be paid of given color\n",
+    "- `card_ability`\n",
+    "    - `card_id` (Integer FK): corresponding to `titleId` in JSON\n",
+    "    - `ability` (String): ability raw text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def filter_raw_card(raw_card):\n",
+    "    \"Remove undesirable columns, change naming conventions and change index to 'titleId'\"\n",
+    "    # Columns to ignore and corresponding reasoning\n",
```

Comment:
What is the biggest list, the list of columns to ignore or the list of allowed columns?
What happens if the site adds a new field in its json? Are you automatically add the column to your table?

Reply:
> What is the biggest list, the list of columns to ignore or the list of allowed columns?

The ignore list is 9 columns bigger than the allowed list.

> What happens if the site adds a new field in its json? Are you automatically add the column to your table?

Functionally, this is an intermediate table before normalization. The tables to be written on a DB will then pick the relevant columns and applying the relevant transformations.

Semantically, I can see that being weird. I've changed the flow to 'pick then pick' instead of 'ignore then pick'